src/Mimi.jl

```diff
@@ -192,11 +192,18 @@ function setindex{T}(m::Model, name::Symbol, values::Vector{T})
     nothing
 end
 
-function addcomponent(m::Model, t, name::Symbol;before=nothing,after=nothing)
+"""
+Add a component to a model.
+"""
+function addcomponent(m::Model, t, name::Union{Symbol,Void}=nothing; before=nothing,after=nothing)
```

Comment:
I found another way for this that is way more elegant and type-stable (not that that matters here)! Try
````julia
function addcomponent(m::Model, t, name::Symbol=Symbol(string(t)); before=nothing,after=nothing)
````
and then you can delete the below ``if`` clause entirely. Sorry for the back and forth, but I just discovered that this works myself.

Oh, and the ``symbol`` function is deprecated and one should use ``Symbol`` instead starting with julia 0.4.